### DISCUSSION TITLE
discuss: using the same donationToken for matchingToken

### DISCUSSION THREAD

**Discussion by thelostone-mc** [dcgtc/dgrants#88] [Architecture & Design] [answered]
#### Description

Based on changes implemented as a result of https://github.com/dcgtc/dgrants/issues/63 

> When deploying a round, you need to include an extra argument for the matching token. (It would probably be good to make this something other than GTC in our dev environment data, so we are properly testing the real-world scenario of different matching vs. donation tokens).

I'm curious so as to why we made this change 
Wouldn't the matchingToken be the same as the donationToken always ?
The distribution generated by dcurve would be the same as the donationToken and translating that to a matchingToken might make things tricky when they are difficult 

Example: 
GrantRoundManager A -> uses GTC as donationToken

If I were to use GrantRoundManager A but specify the matchingToken as DOGE -> the distribution generated by dcurve would be in GTC and not doge.

Is there a benefit on having the donationToken different from matchingToken ?

@apbendi / @gdixon / @corydickson

**Comment by bobjiang** [comment]
Hi, 

As I know, when someone donated on Gitcoin, he/she could donate with any ERC20 token. 
Correct me if I am wrong.

So if it is yes, for matchingToken it is hard to match any ERC20 token from the donor.
But if Gitcoin choose DAI/USDC/GTC, it makes sense.

Not sure if it is this case.

**Comment by mds1** [comment, marked as answer]
So the reason for this is because we don't want to force round operators to payout match amounts in the `donationToken` specified by the `GrantRoundManager`. For example, if `GrantRoundManager` uses GTC as the donation token, all donated funds are converted to GTC and sent to the grant owners. Now if Badger wants to run it's own grants round, they probably want to pay match funds in BADGER. But if we don't allow that, they'll have to sell all the BADGER for GTC and they won't want to do that. Worst case is they create a new `GrantRoundManager` with a different donation token, which fragments composability

> Example:
> GrantRoundManager A -> uses GTC as donationToken
> 
> If I were to use GrantRoundManager A but specify the matchingToken as DOGE -> the distribution generated by dcurve would be in GTC and not doge.

(I'll continue to use BADGER here instead of DOGE, but same idea)

What we want the match calculations to output is not exact match amounts, but ratios of what each grant should get relative to the total matching pool. (And this can be easily computed even if it currently outputs exact token match amounts). This way, from those ratios, you can feed in the GTC donation data, compute the match ratios with dcurve, then compute how much BADGER each grant should be receiving based on the ratios. You could even specify input token/output token as parameters to dcurve, and have it compute BADGER amounts for you
